docs/apps/docker.md

```diff
@@ -82,73 +91,210 @@ This tutorial also provides valuable troubleshooting insights and advice on what
+to preempt the download of your favorite templates in every run:
+
+``` {.shell hl_lines="5 6"}
 $ docker run -ti --rm \
     -v path/to/data:/data:ro \
     -v path/to/output:/out \
+    -v path/to/work:/work \
+    -v path/to/tf-cache:/opt/templateflow \  # mount from host
+    -e TEMPLATEFLOW_HOME=/opt/templateflow \ # override TF home
     nipreps/fmriprep:<latest-version> \
     /data /out/out \
     participant
+    -w /work
+```
+
+!!! warning "*Docker for Windows* requires enabling Shared Drives"
+
+    On *Windows* installations, the `-v` argument will not work
+    by default because it is necessary to enable shared drives.
+    Please check on this [Stackoverflow post](https://stackoverflow.com/a/51822083) how to enable them.
+
+### Running containers as a user
+By default, Docker will run the container with the
+user id (uid) **0**, which is reserved for the default **root**
+account in *Linux*.
+In other words, by default *Docker* will use the superuser account
+to execute the container and will write files with the corresponding
+uid=0 unless configured otherwise.
+Executing as superuser may derive in permissions and security issues,
+for example, [with *DataLad* (discussed later)](datalad.md#).
+One paramount example of permissions issues where beginners typically
+run into is deleting files after a containerized execution.
+If the uid is not overridden, the outputs of a containerized execution
+will be owned by **root** and group **root**.
+Therefore, normal users will not be able to modify the output and
+superuser permissions will be required to deleted data generated
+by the containerized application.
+Some shared systems only allow running containers as a normal user
+because the user will not be able to action on the outputs otherwise.
+
+Either way (whether the container is available with default settings
+or the execution has been customized to normal users),
+running as a normal user allows preempting these permissions issues.
+This can be achieved with
+[*Docker*'s `-u`/`--user` option](https://docs.docker.com/engine/containers/run/#user):
+
+```
+--user=[ user | user:group | uid | uid:gid | user:gid | uid:group ]
 ```
 
-For example: :
+We can combine this option with *Bash*'s `id` command to ensure the current user's uid and group id (gid) are being set:
 
+``` {.shell hl_lines="4"}
+$ docker run -ti --rm \
+    -v path/to/data:/data:ro \
+    -v path/to/output:/out \
+    -u $(id -u):$(id -g) \                   # set execution uid:gid
+    -v path/to/tf-cache:/opt/templateflow \  # mount from host
+    -e TEMPLATEFLOW_HOME=/opt/templateflow \ # override TF home
+    nipreps/fmriprep:<latest-version> \
+    /data /out/out \
+    participant
 ```
+
+For example:
+
+``` Shell
 $ docker run -ti --rm \
     -v $HOME/ds005:/data:ro \
     -v $HOME/ds005/derivatives:/out \
     -v $HOME/tmp/ds005-workdir:/work \
+    -u $(id -u):$(id -g) \
+    -v $HOME/.cache/templateflow:/opt/templateflow \
```

Comment:
mightn't it cause issues in case of parallel execution where multiple runs would compete over?

Reply:
I'll add an admonition about this. Thanks

Reply:
Added in 44360cb. Please send further suggestions or reopen this conversation if that fell short of addressing the issue.